### ISSUE TITLE
Fee percentage calculating tax wrong

### ISSUE THREAD

**Issue by lewishalliday** [woocommerce/woocommerce#25161]
**Describe the bug**
When we add the fee as a percentage it seems to be calculating the pricing wrong.

It seems to be calculating:
_(Product + Tax) + 10% Fee + Tax again_
Is there a way to set the fee to either be zero tax or just be the product price excl. tax

**To Reproduce**

1. Place order (cash on delivery no payment)
2. Go to orders admin 
3. Click 'Add Fee' 
4. Add a positive/negative percentage.
5. Fee is inclusive of product tax then add's tax on-top.

**Screenshots**
https://imgur.com/u2Lbh6m

**Expected behavior**
Fee to calculate percentage +/- of product/s without tax then add applicable tax on top
or
calculate percentage +/- of product/s with tax and then not apply any tax to the fee.

**Isolating the problem (mark completed items with an [x]):**
- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**WordPress Environment**
<details>
```
`
`
### WordPress Environment ###

WordPress address (URL): http://site.net
Site address (URL): http://site.net
WC Version: 3.8.1
REST API Version: ✔ 1.0.2
Log Directory Writable: ✔
WP Version: 5.3
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: ✔

### Server Environment ###

Server Info: Apache/2.4.29 (Ubuntu)
PHP Version: 7.2.24-0ubuntu0.18.04.1
PHP Post Max Size: 16 MB
PHP Time Limit: 600
PHP Max Input Vars: 1000
cURL Version: 7.58.0
OpenSSL/1.1.1

SUHOSIN Installed: –
MySQL Version: 5.7.28-0ubuntu0.18.04.4
Max Upload Size: 16 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 3.8.1
WC Database Prefix: wp_
MaxMind GeoIP Database: ✔
Total Database Size: 18.52MB
Database Data Size: 16.32MB
Database Index Size: 2.20MB
wp_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.06MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_aodfw_coupons: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_aodfw_rules: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_aodfw_spins: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_aodfw_spins_log: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_et_divi_ab_testing_stats: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_ewwwio_images: Data: 0.31MB + Index: 0.23MB + Engine InnoDB
wp_ewwwio_queue: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_nm_personalized: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_options: Data: 2.33MB + Index: 0.13MB + Engine InnoDB
wp_postmeta: Data: 1.52MB + Index: 0.39MB + Engine InnoDB
wp_posts: Data: 10.48MB + Index: 0.19MB + Engine InnoDB
wp_smush_dir_images: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_spellcheck_dictionary: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_spellcheck_empty: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_spellcheck_grammar: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_spellcheck_grammar_options: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_spellcheck_html: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_spellcheck_ignore: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_spellcheck_options: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_spellcheck_words: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.06MB + Index: 0.03MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_w3tc_cdn_pathmap: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_w3tc_cdn_queue: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wccs_conditions: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wccs_condition_meta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wcfm_daily_analysis: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wcfm_detailed_analysis: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wcfm_enquiries: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wcfm_following_followers: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wcfm_messages: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wcfm_messages_modifier: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wcfm_messages_stat: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wcfm_support: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wcfm_support_response: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woof_query_cache: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wpfront_ure_options: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpml_mails: Data: 0.20MB + Index: 0.00MB + Engine InnoDB
wp_yith_wapo_groups: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_yith_wapo_types: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_yoast_seo_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_yoast_seo_meta: Data: 0.02MB + Index: 0.00MB + Engine InnoDB

### Security ###

Secure connection (HTTPS): ❌
					Your store is not using HTTPS. Learn more about HTTPS and SSL Certificates.
Hide errors from visitors: ✔

### Active Plugins (17) ###

All-in-One WP Migration S3 Extension: by ServMask – 3.51
All-in-One WP Migration Unlimited Extension: by ServMask – 2.32
All-in-One WP Migration: by ServMask – 7.11
Better Search Replace: by Delicious Brains – 1.3.3
EWWW Image Optimizer: by Exactly WWW – 5.1.3
Kadence WooCommerce Email Designer: by Kadence WP – 1.3.5
Really Simple SSL: by Rogier Lankhorst
Mark Wolters – 3.2.6

Safe SVG: by Daryll Doyle – 1.9.6
W3 Total Cache: by BoldGrid – 0.11.0
WooCommerce Better Customer List: by Blaze Concepts – 1.2.1 – Not tested with the active version of WooCommerce
Extra Product Options (Product Addons) for WooCommerce: by ThemeHigh – 2.0.7 – Not tested with the active version of WooCommerce
Product Attachment for WooCommerce: by Thedotstore – 1.3.0 – Not tested with the active version of WooCommerce
NextMove Lite - Thank You Page for WooCommerce: by XLPlugins – 2.11.0
WooCommerce Checkout Manager: by QuadLayers – 4.7.8 – Not tested with the active version of WooCommerce
WooCommerce: by Automattic – 3.8.1
Yoast SEO: by Team Yoast – 12.6.2
WP Mail Logging: by MailPoet – 1.9.5

### Inactive Plugins (1) ###

WOOF - WooCommerce Products Filter: by realmag777 – 1.2.3 – Not tested with the active version of WooCommerce

### Dropin Plugins (3) ###

advanced-cache.php: advanced-cache.php
db.php: db.php
object-cache.php: object-cache.php

### Must Use Plugins (1) ###

ET Support Center :: Must-Use Plugins Autoloader: by Elegant Themes –

### Settings ###

API Enabled: ✔
Force SSL: ✔
Currency: GBP (£)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –

### WC Pages ###

Shop base: #251 - /quote/
Cart: #4 - /cart/
Checkout: #5 - /checkout/
My account: #6 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: site
Version: – 0 is available
Author URL: 
Child Theme: ✔
Parent Theme Name: Divi
Parent Theme Version: 4.0.7
Parent Theme Author URL: http://www.elegantthemes.com
WooCommerce Support: ✔

### Templates ###

Overrides: site/woocommerce/emails/admin-new-order.php version 3.5.0 is out of date. The core version is 3.7.0
site/woocommerce/emails/email-addresses.php
site/woocommerce/emails/email-order-details.php version 3.3.1 is out of date. The core version is 3.7.0
site/woocommerce/myaccount/dashboard.php
site/woocommerce/myaccount/my-orders.php
site/woocommerce/myaccount/orders.php version 3.2.0 is out of date. The core version is 3.7.0
site/woocommerce/myaccount/view-order.php
site/woocommerce/order/order-details.php version 3.5.2 is out of date. The core version is 3.7.0

Outdated Templates: ❌
					
					
						Learn how to update

### Action Scheduler ###

Complete: 5
Oldest: 2019-11-04 09:35:03 +0000
Newest: 2019-11-25 17:20:25 +0000

Pending: 0
Oldest: –
Newest: –

Canceled: 2
Oldest: 2019-11-25 12:56:51 +0000
Newest: 2019-11-25 12:58:25 +0000

In-progress: 0
Oldest: –
Newest: –

Failed: 4
Oldest: 2019-06-14 20:36:27 +0000
Newest: 2019-06-14 20:36:27 +0000

`
`
```
</details>

**Comment by juliaamosova** [comment]
Thanks for reporting the issue @lewishalliday. I can reproduce it on my end:

![add_fee](https://user-images.githubusercontent.com/19143190/70174103-336ad400-16cc-11ea-9ef2-aade1197a922.jpg)

The best way to handle it will be to offer a choice to the user whether to apply tax to the fee or not. We are already doing it with shipping fee (it can be taxable or not). 

We can add an option of the fee to be taxable / non-taxable to the dialog when we the user is adding the fee:

![taxable](https://user-images.githubusercontent.com/19143190/70174682-47fb9c00-16cd-11ea-9a78-6d29051b9df3.jpg)

**Comment by lewishalliday** [comment]
Hi @juliaamosova ,

Thank you for the reply and looking at this issue.

The solution of taxable/non-taxable in the dialog box would be a perfect solution in my eyes!

**Comment by juliaamosova** [comment]
The AJAX method that handles adding order fees is: https://github.com/woocommerce/woocommerce/blob/cd4b2ac5b8a9d7053962bd7eebba37639746a00c/includes/class-wc-ajax.php#L953-L1013

**Comment by juliaamosova** [comment]
Hi @lewishalliday,

While we appreciate you sharing your ideas with us, it doesn’t fit in with our current priorities for the project. At some point, we may revisit our priorities and look through the list of suggestions like this one to see if it warrants a second look. In the meantime, we are going to close this issue with the `votes needed` label and evaluate over time if this issue collects more feedback. 

Don’t be alarmed if you don’t see any activity on this issue for a while.

**Comment by woocommercebot** [comment]
Thanks for the suggestion @lewishalliday.

We'll keep an eye on the [popularity of this request](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md#feature-requests) :)
